### ISSUE TITLE
12$: warn on '} if' on same line with no 'else'

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1217]
```javascript
if (x) {
} if (y) {  // warn please!
}
```
```javascript
if (x) { } if (y) { }  // unsure if we should warn...
```
Inspiration: https://twitter.com/tsoding/status/1782676054093533328

**Comment by strager** [comment]
Proposed logic:

After parsing the `{` `}` body of an `if`, check if the next token is `Token_Type::kw_if` and if `has_leading_newline` is false. If yes, then warn.

**Comment by qnguyen1013** [comment]
Hey Strager, I'm a recent cs college grad and am trying to get into open source, would love to try and tackle this issue!

**Comment by strager** [comment]
@qnguyen1013 Sure thing! Here are our contribution guides: https://quick-lint-js.com/contribute/

If you need any help, let me know.

**Comment by singalhimanshu** [comment]
hi @strager would like to take this issue if no one else is actively working on it.

**Comment by NilayPatel10721** [comment]
hi @strager  please assigned me this ,i want to work on this
